openff/interchange/operations/_combine.py

```diff
@@ -147,6 +151,15 @@ def _combine(
                     _tmp_pot_key.mult = _mult
                     _mult += 1
 
+            if new_top_key in self_collection.key_map:
+                raise UnsupportedCombinationError(
+                    "Key collision detected, please raise an issue",
+                )
+
```

Comment:
I don't think this can be triggered - keys are modified to represent _new_ atom indices - so I'm removing it